### DISCUSSION TITLE
Astro init Beer JS

### DISCUSSION THREAD

**Discussion by gw2princeps** [beercss/beercss#127] [Q&A] [answered]
Hey, 

I'm working on a website with Astro and Beercss. I want to make use of a Tabs component, however, I can't figure out how to initialize the JS. 

This very simple example doesnt seem to work - the tabs are not interactable. Also tried to call `window.ui()` in the console, no luck. The script block in the component below is executed on the client side. 

Component.astro
```
---
---

<script>
  import beercss from "beercss";

  beercss("setup");
</script>

<div>
  <div class="tabs">
    <a class="active">Tab 1</a>
    <a>Tab 2</a>
    <a>Tab 3</a>
  </div>

  <div class="page active">
    <h5>Tab 1</h5>
  </div>
  <div class="page">
    <h5>Tab 2</h5>
  </div>
  <div class="page">
    <h5>Tab 3</h5>
  </div>
</div>
```

Additionally, I tried to inline the script, but also no luck. 
```
<script
  is:inline
  type="module"
  src="https://cdn.jsdelivr.net/npm/beercss@3.0.8/dist/cdn/beer.min.js"
></script>
```

Also tried different versions since the examples I found were using 3.0.2, but nothing changed. Seems like I am missing something. Maybe anyone got an idea?

**Comment by leonardorafael** [comment]
Hi,

All elements have a section called trigger.
https://github.com/beercss/beercss/blob/main/docs/TABS.md

The method with data-ui attribute is the most simple, it has the same logic
in all elements and no need to write JS code, just HTML.

All best

Em qui., 23 de fev. de 2023 às 08:59, Princeps ***@***.***>
escreveu:

**Comment by gw2princeps** [comment, marked as answer]
Hey, thanks for your reply! 

I reviewed my code and found the issue. 

The correct way to import the `beer.min.js` file in an Astro Component works as follows:
```html
<script src="beercss/dist/cdn/beer.min.js"></script>

<div class="tabs">
  <a data-ui="#page1">Tab 1</a>
  <a data-ui="#page2" class="active">Tab 2</a>
  <a data-ui="#page3">Tab 3</a>
</div>

<div class="page" id="page1">
  <h5>Tab 1</h5>
</div>
<div class="page" id="page2">
  <h5>Tab 2</h5>
</div>
<div class="page" id="page3">
  <h5>Tab 3</h5>
</div>
```

Have a nice day!
